src/main/java/gift/exception/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,59 @@
+package gift.exception;
+
+import java.time.LocalDateTime;
+import java.util.HashMap;
+import java.util.Map;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.FieldError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+@ControllerAdvice
+public class GlobalExceptionHandler {
+
+    @ResponseBody
+    @ExceptionHandler(ProductNotFoundException.class)
+    public ResponseEntity<ExceptionResponseDto> handleProductNotFoundException(ProductNotFoundException e) {
+        ExceptionResponseDto exception = new ExceptionResponseDto(e.getMessage(), 404, LocalDateTime.now());
+
+        return ResponseEntity.status(HttpStatus.NOT_FOUND).body(exception);
+    }
+
+    @ResponseBody
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    public ResponseEntity<ExceptionResponseDto> handleMethodArgumentNotValidException(MethodArgumentNotValidException e) {
+        StringBuilder sb = new StringBuilder();
+
+        for (FieldError fieldError : e.getBindingResult().getFieldErrors()) {
+            sb.append(fieldError.getField())
+                .append(":")
+                .append(fieldError.getDefaultMessage());
+        }
+
+        ExceptionResponseDto exception = new ExceptionResponseDto(
+            sb.toString(),
+            400,
+            LocalDateTime.now()
+        );
+
+        return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(exception);
```

Comment:
아래와 같이 변경해보는건 어떨까요? 🤗 

```suggestion
        String errorMessage = e.getBindingResult().getFieldErrors()
            .stream()
            .map(fieldError -> fieldError.getField() + ":" + fieldError.getDefaultMessage())
            .collect(Collectors.joining(", "));

        ExceptionResponseDto exception = new ExceptionResponseDto(
            errorMessage,
            HttpStatus.BAD_REQUEST.value(),
            LocalDateTime.now()
        );
```